core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -1197,7 +1197,8 @@ private boolean checkConsistencyInternal(LockedInodePath inodePath) throws Inval
         return ufs.isDirectory(ufsPath);
       } else {
         return ufs.isFile(ufsPath)
-            && ufs.getFileStatus(ufsPath).getContentLength() == inode.asFile().getLength();
+            && ufs.getFileStatus(ufsPath).getContentLength() == inode.asFile().getLength()
+            && (!ufs.getFingerprint(ufsPath).equals(Constants.INVALID_UFS_FINGERPRINT));
```

Comment:
we should compare `ufs.getFingerprint(ufsPath)` with `inode.asFile().getUfsFingerprint` if both are valid

Reply:
actually, this can be made more efficient by combining the ufs calls into ufs.getStatus:

ufsIsDirectory = ufsStatus.isDirectory()
ufsIfFile = ufsStatus.isFile()
ufsFingerprint = Fingerprint.create(ufs.getUnderFSType(), ufsStatus).serialize();

Reply:
@madanadit I have tried to use the ufsStatus in the below way 
`try (CloseableResource<UnderFileSystem> ufsResource = resolution.acquireUfsResource()) {
      UnderFileSystem ufs = ufsResource.get();
      String ufsPath = resolution.getUri().getPath();
      UfsStatus ufsStatus = ufs.getStatus(ufsPath);
      if (ufs == null) {
        return true;
      }
      if (!inode.isPersisted()) {
        return !ufs.exists(ufsPath);
      }
      // TODO(calvin): Evaluate which other metadata fields should be validated.
      if (inode.isDirectory()) {
        return ufsStatus.isDirectory();
      } else {
        String ufsFingerprint = Fingerprint.create(ufs.getUnderFSType(), ufsStatus).serialize();
        return ufsStatus.isFile()
            && ufs.getFileStatus(ufsPath).getContentLength() == inode.asFile().getLength()
            && (ufsFingerprint.equals(inode.asFile().getUfsFingerprint()));
      }`

Reply:
`try (CloseableResource<UnderFileSystem> ufsResource = resolution.acquireUfsResource()) {
      UnderFileSystem ufs = ufsResource.get();
      String ufsPath = resolution.getUri().getPath();
      UfsStatus ufsStatus = ufs.getStatus(ufsPath);
      if (ufs == null) {
        return true;
      }
      if (!inode.isPersisted()) {
        return !ufs.exists(ufsPath);
      }
      // TODO(calvin): Evaluate which other metadata fields should be validated.
      if (inode.isDirectory()) {
        return ufsStatus.isDirectory();
      } else {
        String ufsFingerprint = Fingerprint.create(ufs.getUnderFSType(), ufsStatus).serialize();
        return ufsStatus.isFile()
            && ufs.getFileStatus(ufsPath).getContentLength() == inode.asFile().getLength()
            && (ufsFingerprint.equals(inode.asFile().getUfsFingerprint()));
      }`

Reply:
> try (CloseableResource<UnderFileSystem> ufsResource = resolution.acquireUfsResource()) {
      UnderFileSystem ufs = ufsResource.get();
      String ufsPath = resolution.getUri().getPath();
      UfsStatus ufsStatus = ufs.getStatus(ufsPath);
      if (ufs == null) {
        return true;
      }
      if (!inode.isPersisted()) {
        return !ufs.exists(ufsPath);
      }
      // TODO(calvin): Evaluate which other metadata fields should be validated.
      if (inode.isDirectory()) {
        return ufsStatus.isDirectory();
      } else {
        String ufsFingerprint = Fingerprint.create(ufs.getUnderFSType(), ufsStatus).serialize();
        return ufsStatus.isFile()
            && ufs.getFileStatus(ufsPath).getContentLength() == inode.asFile().getLength()
            && (ufsFingerprint.equals(inode.asFile().getUfsFingerprint()));
      }

Reply:
@tahakashaf can you please update the PR. you can move the getStatus call to after the null check for ufs and catch FileNotFoundException from ufs.getStatus instead of the ufs.exists call